### PR TITLE
Feature plugin systerm

### PR DESCRIPTION
**Pull request by languitar**
Prepares for separating the spread transport by implementing a plugin system that is used for registering transports.

### CODE REVIEW

File: rsb-java/src/main/java/rsb/plugin/LoadingException.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ */
+
+package rsb.plugin;
+
+/**
+ * Represent an error while loading a specific plugin.
+ *
+ * Instances have to be default-constructible.
+ *
+ * @@author jwienke
```
Comment (scymtym) [inline rsb-java/src/main/java/rsb/plugin/LoadingException.java]: Should this be `@author`?

File: rsb-java/src/main/java/rsb/plugin/LoadingException.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * ============================================================
+ */
+
+package rsb.plugin;
+
+/**
+ * Represent an error while loading a specific plugin.
+ *
+ * Instances have to be default-constructible.
```
Comment (scymtym) [inline rsb-java/src/main/java/rsb/plugin/LoadingException.java]: I don't understand this constraint.

Comment (languitar) [reply rsb-java/src/main/java/rsb/plugin/LoadingException.java]: default constructor in Java is one without arguments so that class.newInstance() works.

Comment (scymtym) [reply rsb-java/src/main/java/rsb/plugin/LoadingException.java]: Sure, I mean, instances of which class?

Comment (languitar) [reply rsb-java/src/main/java/rsb/plugin/LoadingException.java]: Opps. The comment ended up in the wrong class.

**Pull request merged**